### PR TITLE
Major UI Additions, Changes & Fixes

### PR DESCRIPTION
**Pull request by ayushyamitabh**
Thank @edwin0108  for the help on these pages  

Create Class - completed, has email validation, error handling, and proper field hierarchy
image

----  

Join Class - redone to better handle errors, and provide a more intuitive ui, shows more detail, and makes traversing available classes easier  

image

image

----  

Profile - "Your Academics" section's UI improved to better use space
image

### CODE REVIEW

**Comment by khelif96** [comment]
Can you also write some tests for this.

**Comment by edwin0108** [comment]
> Can you also write some tests for this.

I'll work with Ayushya with the unit testing.

File: website/package.json [hunk 1]
```diff
@@ -49,5 +49,10 @@
     "react-test-renderer": "16.5.2",
     "stylelint": "9.6.0",
     "stylelint-config-standard": "18.2.0"
+  },
+  "eslintConfig": {
```
[the remark this answers is not in the document]
Comment (ayushyamitabh) [reply website/package.json]: Not sure, I think it was from @edwin0108 ? Or maybe from merging from master

File: website/src/App.css [hunk 2]
```diff
@@ -17,3 +17,9 @@
   height: auto !important;
   color: white !important;
 }
+
+.profile-loading svg {
+  width: 50px !important;
```
[the remark this answers is not in the document]
Comment (ayushyamitabh) [reply website/src/App.css]: Using the `!important` tag is the proper way to override css attributes supplied by design libraries. Ant Design is however inconsistent in how they implement their styles, some styles seem to be in line and can be overriden without the tag.

File: website/src/views/CreateClass/CreateClass.css [hunk 3]
```diff
@@ -26,8 +26,45 @@
...
+
+.create-class-days-label {
+  margin: 0;
+  margin-top: 15px !important;
+  margin-bottom: 5px !important;
+  font-size: 16px !important;
+}
+
+.create-class-days-hint {
+  margin: 0;
```
[the remark this answers is not in the document]
Comment (ayushyamitabh) [reply website/src/views/CreateClass/CreateClass.css]: Could've done `margin: 2.5 auto;` but left and right margins have values that I don't want to change.

File: website/src/views/JoinClass/JoinClass.jsx [hunk 4]
```diff
@@ -35,181 +37,217 @@ class JoinClass extends Component {
...
+              <span>Instructor: </span>
+              {selectedClass.instructorName}
+            </p>
+            <p className="join-class-label">
+              <span>Time: </span>
+              {selectedClass.time}
+            </p>
+            <p className="join-class-label">
+              <span>Day(s): </span>
+              {selectedClass.days}
```
[the remark this answers is not in the document]
Comment (ayushyamitabh) [reply website/src/views/JoinClass/JoinClass.jsx]: Yes, it has to do with the API data being returned. It will fix itself when API is updated.

**Comment by ayushyamitabh** [comment]
@khelif96
The only reused part was the college search component, we can make that into a separate component but I don't think we need a new package for this.

@edwin0108 can you start with writing tests for these components, I will fix APIs and then return to this cause.

@whuang001 can you look into the documentation for PWA service worker from Google and we can then start work on creating a some sort of middleware for fetching API data along with custom cache management

**Comment by khelif96** [comment]
Sorry i should clarify When i say reuse i mean the repeated configuration for  whenever we create a form. Declaring a state for each form value, Setting up onchange listeners Doing form validation etc. This library does all of that for us

**Comment by whuang001** [comment]
> @khelif96
> The only reused part was the college search component, we can make that into a separate component but I don't think we need a new package for this.
> 
> @edwin0108 can you start with writing tests for these components, I will fix APIs and then return to this cause.
> 
> @whuang001 can you look into the documentation for PWA service worker from Google and we can then start work on creating a some sort of middleware for fetching API data along with custom cache management

Sure, I will be work on that. Also, can I help you with the API stuff? i.e. forgot password...

File: website/src/views/CreateClass/CreateClass.jsx [hunk 5]
```diff
@@ -34,106 +34,203 @@ class CreateClass extends Component {
...
+        instructorName: firebase.auth().currentUser.displayName,
+        approvedEmails,
+        meetingTimes: {
+          from: fromTime,
+          to: toTime,
+        },
+        meetingDays,
       },
     };
+    console.log(reqData);
```
[the remark this answers is not in the document]
Comment (ayushyamitabh) [reply website/src/views/CreateClass/CreateClass.jsx]: fixed in next commit

File: website/src/views/Profile/Profile.jsx [hunk 6]
```diff
@@ -75,46 +85,68 @@ class Profile extends PureComponent {
...
+                                        <Button icon="close" size="small" shape="circle" type="danger" />,
+                                      ]}
+                                    >
+                                      {item}
+                                    </List.Item>
+                                  )}
+                                />
+                              </Collapse.Panel>
+                            );
+                            // <Collapse.Panel header={university} key={university}>
```
[the remark this answers is not in the document]
Comment (ayushyamitabh) [reply website/src/views/Profile/Profile.jsx]: fixed in next commit

File: website/src/views/Profile/Profile.jsx [hunk 7]
```diff
@@ -123,7 +155,7 @@ class Profile extends PureComponent {
               width: '100%', height: '100%', display: 'flex', justifyContent: 'center', alignItems: 'center',
```
[the remark this answers is not in the document]
Comment (ayushyamitabh) [reply website/src/views/Profile/Profile.jsx]: fixed in next commit

File: website/src/views/JoinClass/JoinClass.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (ayushyamitabh) [reply website/src/views/JoinClass/JoinClass.jsx]: fixed in next commit

**Pull request merged**